### PR TITLE
Dev business

### PR DESCRIPTION
**Pull request by SashaDrapun**
Display and delete for members and tasks. Also create for tasks.

### CODE REVIEW

File: DIMS-Core/DIMS-Core.BusinessLayer/MappingProfiles/MemberProfile.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+using EntityUserProfile = DIMS_Core.DataAccessLayer.Entities.UserProfile;
+
+namespace DIMS_Core.BusinessLayer.MappingProfiles
+{
+    internal class MemberProfile : Profile
+    {
+        public MemberProfile()
+        {
+            CreateMap< EntityUserProfile, UserProfileModel>()
+                .ForMember(dest => dest.Email,
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/MappingProfiles/MemberProfile.cs:16]: This mapping for Email is redundant.

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/TaskService.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        public async Task CreateAsync(TaskModel model)
+        {
+            if (model is null || model.TaskId != 0)
+            {
+                return;
+            }
+
+            var entity = mapper.Map<DataAccessLayer.Entities.Task>(model);
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.BusinessLayer/Services/TaskService.cs:55]: Remove namespace please and rename type Task by keyword using.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs [hunk 3]
```diff
@@ -33,7 +35,7 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
             if (!optionsBuilder.IsConfigured)
             {
 #warning To protect potentially sensitive information in your connection string, you should move it out of source code. See http://go.microsoft.com/fwlink/?LinkId=723263 for guidance on storing connection strings.
-                optionsBuilder.UseSqlServer("Server=DESKTOP-N10FT1S;Database=DIMS;Trusted_Connection=True;");
+                optionsBuilder.UseSqlServer(configuration.GetConnectionString("DIMSDatabase"));
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs:38]: Move this in startup.

File: DIMS-Core/DIMS-Core/Models/Member/EditMemberViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        public int UserId { get; set; }
+        public string Name { get; set; }
+        public string LastName { get; set; }
+        public string Education { get; set; }
+        public double? UniversityAverageScore { get; set; }
+        public double? MathScore { get; set; }
+        public string Address { get; set; }
+        public string MobilePhone { get; set; }
+        public string Skype { get; set; }
+    }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Models/Member/EditMemberViewModel.cs:20]: Where are validation attributes?

File: DIMS-Core/DIMS-Core/Models/Member/MemberViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class MemberViewModel
+    {
+        public int UserId { get; set; }
+        public string FullName { get; set; }
+        public string Direction { get; set; }
+        public string Education { get; set; }
+        public int? Age { get; set; }
+        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:dd.MM.yyyy}")]
+        public DateTime? StartDate { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Models/Member/MemberViewModel.cs:18]: Validation attributes?

File: DIMS-Core/DIMS-Core/Models/Task/TaskViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+{
+    public class TaskViewModel
+    {
+        public int TaskId { get; set; }
+        public string Name { get; set; }
+        public string Description { get; set; }
+        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:dd.MM.yyyy}")]
+        public DateTime StartDate { get; set; }
+        [DisplayFormat(ApplyFormatInEditMode = true, DataFormatString = "{0:dd.MM.yyyy}")]
+        public DateTime DeadlineDate { get; set; }
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Models/Task/TaskViewModel.cs:18]: Validation attributes?

File: DIMS-Core/DIMS-Core/Views/Member/Create.cshtml [hunk 7]
```diff
@@ -0,0 +1,82 @@
+﻿@model AddMemberViewModel
+
+@{
+    ViewData["Title"] = "View";
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Views/Member/Create.cshtml:4]: View?

File: DIMS-Core/DIMS-Core/Views/Member/Create.cshtml [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+<h1>View</h1>
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Views/Member/Create.cshtml:7]: View?

File: DIMS-Core/DIMS-Core/Views/Member/Create.cshtml [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+<h4>MemberModel</h4>
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Views/Member/Create.cshtml:9]: MemberModel?

File: DIMS-Core/DIMS-Core/Views/Member/Delete.cshtml [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+@{
+    ViewData["Title"] = "Delete";
+}
+
+<h1>Delete</h1>
+
+<h3>Are you sure you want to delete this?</h3>
+<div>
+    <h4>MemberModel</h4>
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Views/Member/Delete.cshtml:11]: MemberModel?

File: DIMS-Core/DIMS-Core/Views/Member/Edit.cshtml [hunk 11]
```diff
@@ -0,0 +1,63 @@
+﻿@model EditMemberViewModel
+
+@{
+    ViewData["Title"] = "Edit";
+}
+
+<h1>Edit</h1>
+
+<h4>MemberModel</h4>
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Views/Member/Edit.cshtml:9]: MemberModel?

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
You did pretty good PR. There are some little issues. Fix them and I'll merge your changes.

File: DIMS-Core/DIMS-Core.BusinessLayer/Services/TaskService.cs (hunk 2 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.BusinessLayer/Services/TaskService.cs]:
```suggestion
            var entity = mapper.Map<TaskEntities>(model);
```

File: DIMS-Core/DIMS-Core.BusinessLayer/MappingProfiles/MemberProfile.cs (hunk 1 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.BusinessLayer/MappingProfiles/MemberProfile.cs:16]: image

File: DIMS-Core/DIMS-Core/Views/Member/Edit.cshtml (hunk 11 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Views/Member/Edit.cshtml:9]: H4 removed from layout

File: DIMS-Core/DIMS-Core/Views/Member/Delete.cshtml (hunk 10 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Views/Member/Delete.cshtml:11]: H4 removed from layout

File: DIMS-Core/DIMS-Core/Views/Member/Create.cshtml (hunk 9 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Views/Member/Create.cshtml:9]: H4 removed from layout

File: DIMS-Core/DIMS-Core/Views/Member/Create.cshtml (hunk 7 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Views/Member/Create.cshtml:4]: image

File: DIMS-Core/DIMS-Core/Views/Member/Create.cshtml (hunk 8 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Views/Member/Create.cshtml:7]: image

File: DIMS-Core/DIMS-Core/Models/Member/EditMemberViewModel.cs (hunk 4 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Models/Member/EditMemberViewModel.cs:20]: Some Validation attributes were added. More validation attributes will be added during development
image

File: DIMS-Core/DIMS-Core/Models/Member/MemberViewModel.cs (hunk 5 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core/Models/Member/MemberViewModel.cs:18]: This model is only used for display data

File: DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs (hunk 3 above)
Comment (PushMeNow) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs:37]: I repeat: move this line in Startup.

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
You didn't resolve one issue about connection string.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs (hunk 3 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs]: image

Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs]: image

File: DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs [hunk 12]
```diff
@@ -2,18 +2,20 @@
...
+using Microsoft.Extensions.Configuration;
 
 namespace DIMS_Core.DataAccessLayer.Context
 {
     public partial class DIMSCoreDatabaseContext : DbContext
     {
-        public DIMSCoreDatabaseContext()
-        {
-        }
+        private readonly IConfiguration configuration;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs:11]: Remove configiration. You don't use it in this class

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hello.
There is a little issue. Look at my comment and fix it.

File: DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs (hunk 12 above)
Comment (SashaDrapun) [reply DIMS-Core/DIMS-Core.DataAccessLayer/Context/DIMSCoreDatabaseContext.cs]: image

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**